tests/unit/sanakoyeu_et_al_2018/test_utils.py

```diff
@@ -0,0 +1,88 @@
+import pytorch_testing_utils as ptu
+import torch
+from torch import nn, optim
+
+import pystiche_papers.sanakoyeu_et_al_2018 as paper
+
+
+def test_optimizer_modules(subtests):
+    transformer = nn.Conv2d(3, 3, 1)
+    params = tuple(transformer.parameters())
+
+    for optimizer_params in [transformer, transformer.parameters()]:
+        with subtests.test("parameters"):
+            optimizer = paper.optimizer(optimizer_params)
+
+            assert isinstance(optimizer, optim.Adam)
+            assert len(optimizer.param_groups) == 1
+
+            param_group = optimizer.param_groups[0]
+
+            with subtests.test(msg="optimization params"):
+                assert len(param_group["params"]) == len(params)
+                for actual, desired in zip(param_group["params"], params):
+                    assert actual is desired
+
+            with subtests.test(msg="optimizer properties"):
+                assert param_group["lr"] == ptu.approx(2e-4)
+
+
+def test_lr_scheduler():
+    transformer = nn.Conv2d(3, 3, 1)
+    optimizer = paper.optimizer(transformer)
+    lr_scheduler = paper.lr_scheduler(optimizer)
+
+    assert isinstance(lr_scheduler, paper.DelayedExponentialLR)
+
+
+def test_ExponentialMovingAverageMeter(subtests):
+    vals = [0.6, 0.7, 0.8]
+    init_val = 0.8
+    smoothing_factor = 0.05
+    name = "test_ema"
+    ema = paper.ExponentialMovingAverageMeter(name, init_val)
+
+    with subtests.test("name"):
+        assert ema.name == name
+
+    with subtests.test("local average"):
+        desired = init_val
+        for val in vals:
+            ema.update(val)
+            desired = desired * (1.0 - smoothing_factor) + smoothing_factor * val
+
+        current = ema.global_avg()
+        assert current == ptu.approx(desired)
+
+
+def test_ExponentialMovingAverageMeter_update_with_tensor(subtests):
+    vals = [torch.tensor([0.6]), torch.tensor([0.7]), torch.tensor([0.8])]
+    init_val = 0.8
+    smoothing_factor = 0.05
+    name = "test_ema"
+    ema = paper.ExponentialMovingAverageMeter(name, init_val)
+
+    with subtests.test("name"):
+        assert ema.name == name
+
+    with subtests.test("init_val"):
+        assert ema.global_avg() == init_val
```

Comment:
Please do not compare `float`s for equality.
```suggestion
        assert ema.global_avg() == pytest.approx(init_val)
```